File: DensitySVD_sep1_16/R_code_old/density1.r
density1<-function(y,D,S,dimIndex,N,bLog){

  ## y=(1:4000)/1000
  ##out=density1(y,D=c(7,5.4),S=c(14,15),1,20,1)
  ##plot(y,exp(out-max(out)),type='l')
  #dyn.load("./C_code/hyper_2by2_R.so")
  dyn.load("./C_code/functions_hyper_2by2_R.so")
  
  val=0*seq(1:length(y));
  
  for (i in  1:length(y)){
    
  x=y[i];
  alpha=1;
  beta=0;
  D[dimIndex]=x;
  eigenValues=D^2/4;
  d = 0.0
  if(bLog==1){
    hyper0F1_val = .C("hyper_2by2_R",a=1.5,eigenValues,d)[[3]]
    val[i]=(alpha-1)*log(x)+(x*(S[dimIndex]-beta))- N * log(hyper0F1_val)
  }else{
    hyper0F1_val = .C("hyper_2by2_R",a=1.5,eigenValues,d)[[3]]
    val[i]=x^(alpha-1)*exp(x*(S[dimIndex]-beta))/(hyper0F1_val)^N
  } 
  
  }
  #write.table(val,file="./density1.txt")
  return(val)  
  
}

File: DensitySVD_sep1_16/matlab/density1.m
function val=density1(y,D,S,dimIndex,N,bLog)
    %mex mhg.c
    %mex logmhg.c
    %D, S
    %
    %dim2=setdiff([1,2],dimIndex)
    %(d1*S(1,1))-N*(mhg(30,2,[],1.5,d))^N
    
    val=zeros(length(y),1);
    for i = 1:length(y)
        x=y(i);
        alpha=1;
        beta=0;
        D(dimIndex)=x;
        eigenValues=D.^2/4;
        if(bLog==1)
            val(i)=(alpha-1)*log(x)+(x*(S(dimIndex)-beta))- N * logmhg(60,2,[],1.5,eigenValues);
        else
            val(i)=x^(alpha-1)*exp(x*(S(dimIndex)-beta))/(mhg(10,2,[],1.5,eigenValues))^N;
        end
    end 
end